### PR TITLE
Multiple Features: Battery Sensors for Lock/Lock Keypad, Wyze Cam Outdoor, Camera Events via Event Bus, Store UUID in Config Entry, __init__.py cleanup

### PR DESCRIPTION
**Pull request by yoinx**
Multiple Features:

Battery Sensors for Lock/Lock Keypad
Wyze Cam Outdoor
Camera Events via Event Bus
Store UUID in Config Entry
init.py cleanup

### CODE REVIEW

File: custom_components/wyzeapi/__init__.py [hunk 1]
```diff
@@ -34,7 +35,7 @@ async def async_setup(
     hass: HomeAssistant, config: HomeAssistantConfig, discovery_info=None
 ):
     # pylint: disable=unused-argument
-    """Set up the Alexa domain."""
+    """Set up the WyzeApi domain."""
```
Comment (JoshuaMulliken) [inline custom_components/wyzeapi/__init__.py:38]: Haha thank you!

File: custom_components/wyzeapi/__init__.py [hunk 2]
```diff
@@ -87,6 +82,16 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
 
     hass.data.setdefault(DOMAIN, {})
 
+    # Store a uuid for the notification toggle if it doesn't exist to get away from storing it in an ini file
+    if not config_entry.data.get(UUID):
+        _LOGGER.debug("No Existing UUID found for %s", DOMAIN)
```
Comment (JoshuaMulliken) [inline custom_components/wyzeapi/__init__.py:87]: Can we check to see if there is an existing wyze_config.ini file here and load from it if it exists?

File: custom_components/wyzeapi/__init__.py [hunk 3]
```diff
@@ -105,8 +110,8 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
             token,
         )
     except:
-        _LOGGER.error("Wyzeapi: Could not login. Please re-login through integration configuration.")
-        raise ConfigEntryAuthFailed("Unable to login, please re-login.")
+        _LOGGER.error("Wyzeapi: Could not login. Please re-login through integration configuration")
+        raise ConfigEntryAuthFailed("Unable to login, please re-login.") from None
```
Comment (JoshuaMulliken) [inline custom_components/wyzeapi/__init__.py:114]: Why the `from None`?

File: custom_components/wyzeapi/__init__.py [hunk 4]
```diff
@@ -147,10 +134,11 @@ def get_uid():
         device_registry, config_entry.entry_id
     ):
         for identifier in device.identifiers:
+            # domain has to remain here. If it is removed the integration will remove all entities for not being in the mac address list each boot.
             domain, mac = identifier
             if mac not in mac_addresses:
                 _LOGGER.warning(
-                    f"{mac} is not in the mac_addresses list. Removing the entry..."
+                    '%s is not in the mac_addresses list, removing the entry', mac
```
Comment (JoshuaMulliken) [inline custom_components/wyzeapi/__init__.py:141]: Is this just a style choice? Not a problem either way

**Review by JoshuaMulliken — CHANGES_REQUESTED** [review]
There are a couple things that I think should change but overall these features are great!

File: custom_components/wyzeapi/__init__.py (hunk 3 above)
Comment (yoinx) [reply custom_components/wyzeapi/__init__.py:114]: Its just something that the lint configuration from home assistant prefers.

File: custom_components/wyzeapi/__init__.py (hunk 4 above)
Comment (yoinx) [reply custom_components/wyzeapi/__init__.py:141]: The home assistant lint configuration says that is a slight optimization. Basically the lazy formatting only happens if it needs to log. Normal substitutions happen always.

File: custom_components/wyzeapi/__init__.py (hunk 2 above)
Comment (yoinx) [reply custom_components/wyzeapi/__init__.py:87]: We could, but that would require us to keep the ini reading code in place in addition to this code. 

During my testing, it didn't seem like the UUID actually *needs* to be persisted to toggle the notification setting.

File: custom_components/wyzeapi/__init__.py (hunk 1 above)
Comment (yoinx) [reply custom_components/wyzeapi/__init__.py:51]: It needed to be done.

File: custom_components/wyzeapi/__init__.py (hunk 2 above)
Comment (yoinx) [reply custom_components/wyzeapi/__init__.py]: This has hopefully be addressed in the latest commit on this PR.

**Review by JoshuaMulliken — DISMISSED** [review]
LGTM. Great Job!

**Review by JoshuaMulliken — APPROVED** [review]
LGTM. I will roll this into the dev branch and get a beta out

**Comment by JoshuaMulliken** [comment]
Beta is out: https://github.com/JoshuaMulliken/ha-wyzeapi/releases/tag/2021.11.1b1

**Comment by JoshuaMulliken** [comment]
This is in dev. I am closing this so that @yoinx can continue work based on his dev branch while we are testing

**Pull request closed without merging**